extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/sensorJobExecuter/AbstractSensorJobExecutor.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.digitalstrom.internal.lib.sensorJobExecuter;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.config.Config;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.manager.ConnectionManager;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.sensorJobExecuter.sensorJob.SensorJob;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.serverConnection.DsAPI;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.Device;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.deviceParameters.DSID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AbstractSensorJobExecutor} provides the working process to execute implementations of {@link SensorJobs}'s
+ * in the time interval that is set in {@link Config#SENSOR_READING_WAIT_TIME}.
+ * <p>
+ * The follow methods can be overridden by subclasses to implement a execution priority:
+ * <ul>
+ * <li>{@link #addLowPriorityJob(SensorJob)}</li>
+ * <li>{@link #addMediumPriorityJob(SensorJob)}</li>
+ * <li>{@link #addHighPriorityJob(SensorJob)}</li>
+ * </ul>
+ * </p>
+ *
+ * @author Michael Ochel - Initial contribution
+ * @author Matthias Siegele - Initial contribution
+ *
+ */
+public abstract class AbstractSensorJobExecutor {
```

Comment:
did you really have to implement your own Executors? This is usually pretty error prone, so I wonder what reasons you had not to use Executors from Java?